examples/audio-classification/run_audio_classification.py

```diff
@@ -220,6 +221,9 @@ def main():
         token=model_args.token,
     )
 
+    if training_args.sdp_on_bf16:
+        torch._C._set_math_sdp_allow_fp16_bf16_reduction(True)
```

Comment:
It would be easier to do it in the training args post_init (or setup_devices) method I think, WDYT? https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/transformers/training_args.py#L832
Same for diffusion pipelines, we could set this directly there.

Reply:
The diffusion example doesn't use GaudiTrainingArgs. Where do you think the bf16 reduction should be called in this case?

Reply:
Sorry it was a bit unclear, I was referring to the pipelines themselves. I think this should go in the `__init__` of the `GaudiDiffusionPipeline` class: https://github.com/huggingface/optimum-habana/blob/48423761ff3039095f9b4408b73c9e094b302d5c/optimum/habana/diffusers/pipelines/pipeline_utils.py#L118

Reply:
I've made the changes. While with training args the code is simpler, I'm not so sure with the pipelines. Tell me what you think.